src/main/java/com/aws/greengrass/clientdevices/auth/iot/CertificateRegistry.java

```diff
@@ -53,14 +78,20 @@ public Optional<Certificate> getCertificateFromPem(String certificatePem) throws
      *
      * @param certificatePem Certificate PEM
      * @return certificate object
+     *
      * @throws InvalidCertificateException if certificate PEM is invalid
+     * @throws CertificateException - If fails generate certificate from PEM
+     * @throws KeyStoreException - If fails to store the key store into disk
      */
-    public Certificate getOrCreateCertificate(String certificatePem) throws InvalidCertificateException {
+    public Certificate getOrCreateCertificate(String certificatePem) throws InvalidCertificateException,
+            CertificateException, KeyStoreException {
```

Comment:
I'm not certain if we should bubble up an exception for a cert store failure. It's not clear what a higher layer would actually do with this information and actually I think the behavior we want would be to continue functioning in a degraded state (we can still authenticate, but can't refresh certs).
Whereas if we throw here, then we can't even authenticate.

Reply:
Great observation. Love it, I will adjust this